pandora-client-web/src/components/settings/securitySettings.tsx

```diff
@@ -0,0 +1,327 @@
+					<input
+						type='password'
+						id='extend-current-session-password'
+						autoComplete='current-password'
+						value={ password }
+						onChange={ (e) => setPassword(e.target.value) }
+					/>
+				</FormField>
+				<br />
+				<Row>
+					<Button onClick={ hide } disabled={ processing }>Cancel</Button>
+					<Button type='submit' disabled={ processing }>Extend</Button>
+				</Row>
+			</Form>
+		</ModalDialog>
+	);
+}
+
+interface PasswordChangeFormData {
+	oldPassword: string;
+	newPassword: string;
+	newPasswordConfirm: string;
+}
+
+function PasswordChange({ account }: { account: IDirectoryAccountInfo; }): ReactElement {
+	const directoryConnector = useDirectoryConnector();
+	const [invalidPassword, setInvalidPassword] = React.useState('');
+
+	const {
+		formState: { errors, submitCount, isSubmitting },
+		reset,
+		getValues,
+		handleSubmit,
+		register,
+		trigger,
+	} = useForm<PasswordChangeFormData>({ shouldUseNativeValidation: true, progressive: true });
+
+	React.useEffect(() => {
+		if (invalidPassword) {
+			void trigger();
+		}
+	}, [invalidPassword, trigger]);
+
+	const onSubmit = handleSubmit(async ({ oldPassword, newPassword }) => {
+		const passwordSha512Old = await PrehashPassword(oldPassword);
+		const passwordSha512New = await PrehashPassword(newPassword);
+		const { cryptoKey, onSuccess } = await directoryConnector.directMessageHandler.passwordChange(account.username, newPassword);
+
+		const resp = await directoryConnector.awaitResponse('passwordChange', {
+			passwordSha512Old,
+			passwordSha512New,
+			cryptoKey,
+		});
+
+		switch (resp.result) {
+			case 'ok':
+				onSuccess();
+				reset();
+				break;
+			case 'invalidPassword':
+				setInvalidPassword('Invalid password');
+				break;
+			default:
+				AssertNever(resp.result);
+		}
+	});
+
+	return (
+		<fieldset>
+			<legend>Password Change</legend>
+			<Form dirty={ submitCount > 0 } onSubmit={ onSubmit }>
+				<FormField>
+					<label htmlFor='password-change-old'>Old password</label>
+					<input
+						type='password'
+						id='password-change-old'
+						autoComplete='current-password'
+						{ ...register('oldPassword', {
+							required: 'Old password is required',
+						}) }
+					/>
+					<FormFieldError error={ errors.oldPassword } />
+				</FormField>
+				<FormField>
+					<label htmlFor='password-change-new'>New password</label>
+					<input
+						type='password'
+						id='password-change-new'
+						autoComplete='new-password'
+						{ ...register('newPassword', {
+							required: 'New password is required',
+							validate: FormCreateStringValidator(PasswordSchema, 'password'),
+						}) }
+					/>
+					<FormFieldError error={ errors.newPassword } />
+				</FormField>
+				<FormField>
+					<label htmlFor='password-change-new-confirm'>Confirm new password</label>
+					<input
+						type='password'
+						id='password-change-new-confirm'
+						{ ...register('newPasswordConfirm', {
```

Comment:
Per HTML spec the "confirm" field for a new password should _also_ have `autoComplete='new-password'`